nrf_modem/include/nrf_socket.h

```diff
@@ -17,7 +17,22 @@
 extern "C" {
 #endif
 
+#if !defined(__GNUC__) || (__GNUC__ == 0)
 typedef int32_t ssize_t;
+#else
+#include <sys/types.h>
+#ifdef __SES_ARM
+typedef int32_t ssize_t;
+#endif
```

Comment:
How can we have `__SES_ARM` defined when `__GNUC__` is defined?

Reply:
I assumed some author of this code in this repository (see repro at modem v1.5.1) knows the answer. I may me removed, I agree.